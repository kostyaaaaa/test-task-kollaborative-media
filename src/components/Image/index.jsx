import { string } from 'prop-types';
import React from 'react';

const Image = ({ src, className }) => {
  return <img alt="" src={src} className={className} />;
};

Image.defaultProps = {
  className: '',
};

Image.propTypes = {
  className: string,
  src: string.isRequired,
};

export default Image;
