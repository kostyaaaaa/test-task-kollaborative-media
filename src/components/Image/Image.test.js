import renderer from 'react-test-renderer';
import Image from './index';

const component = renderer.create(
  <Image src="image-source" className="text-xl" />,
);

it('should match snapshot', () => {
  expect(component).toMatchSnapshot();
});

it('should apply provided props', () => {
  expect(component.toTree().rendered.props.className).toEqual('text-xl');
  expect(component.toTree().rendered.props.src).toEqual('image-source');
});

it('should have default alt prop', () => {
  expect(component.toTree().rendered.props.alt).toEqual('');
});
