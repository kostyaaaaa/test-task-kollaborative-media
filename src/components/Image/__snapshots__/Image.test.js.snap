// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<img
  alt=""
  className="text-xl"
  src="image-source"
/>
`;
